### PR TITLE
Intermediate1.ATX.2017Q1.Gucciardi.Jonathan

### PR DESCRIPTION
**Pull request by SalamanderVertigo**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of JavaScript: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of jQuery: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [x] Effective use of AJAX: 20 points
- [x] Effective use of loops: 20 points
- [x] Effective use of github API: 20 points
- [x] Style and design: 10 points
- [ ] Properly commented code: 15 points
- [x] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/02RockPaperScissors.js [hunk 1]
```diff
@@ -8,9 +8,24 @@ var rl = readline.createInterface({
 });
 
 
-function rockPaperScissors(hand1, hand2) {
+function rockPaperScissors(hand1, hand2) { // rock paper and scissors are hard coded
 
   // Write code here
+if(hand1 === 'paper' && hand2 === 'rock'){
+  return "Hand1 Wins";
```
Comment (georgewmitchell) [inline apps/02RockPaperScissors.js]: Great work

File: apps/02RockPaperScissors.js [hunk 2]
```diff
@@ -8,9 +8,24 @@ var rl = readline.createInterface({
...
+    return ("Hand1 wins!");
+    }
+if(hand1 === 'rock' && hand2 === 'scissors'){
+  return "Hand1 WINS!";
+    }
+if(hand1 === hand2){
+  return "its a tie";
+}
+else{
+  return "Hand2 Wins";
```
Comment (georgewmitchell) [inline apps/02RockPaperScissors.js]: Super

File: apps/02RockPaperScissors.js [hunk 3]
```diff
@@ -8,9 +8,24 @@ var rl = readline.createInterface({
...
+function rockPaperScissors(hand1, hand2) { // rock paper and scissors are hard coded
 
   // Write code here
+if(hand1 === 'paper' && hand2 === 'rock'){
+  return "Hand1 Wins";
+  }
+if(hand1 === 'scissors' && hand2 === 'paper'){
+    return ("Hand1 wins!");
+    }
+if(hand1 === 'rock' && hand2 === 'scissors'){
```
Comment (georgewmitchell) [inline apps/02RockPaperScissors.js]: rock paper scissors

File: apps/03TicTacToe.js [hunk 4]
```diff
@@ -12,6 +12,26 @@ var board = [
...
+ board = [
+  [' ', ' ', ' '],
+  [' ', ' ', ' '],
+  [' ', ' ', ' ']
+];
+return board;
+}
+
+function checkCell(){
+  var i = 0;
```
Comment (jw56578) [inline apps/03TicTacToe.js]: Double check spelling. Rerun your tests

File: apps/03TicTacToe.js [hunk 5]
```diff
@@ -22,25 +44,71 @@ function printBoard() {
...
+  if(horizontalWin() === true || verticalWin() === true || diagonalWin() === true){
+     console.log('Player ' + playerTurn + ' Won!');
+  resetBoard();
+return true;
+  }
+
+
 }
 
+/* This is where the game will be played. playerTurn is default as 'X', assign it to the board and
```
Comment (pieroapretto) [inline apps/03TicTacToe.js]: Good job leaving lots of comments.

File: apps/04TowersOfHanoi.js [hunk 6]
```diff
@@ -13,30 +13,73 @@ var stacks = {
...
+   var temp; // store the popped stack value here, temp
 
-function movePiece() {
-  // Your code here
+    temp = stacks[startStack].pop(); // pop the user selected stack and store value in temp
+    stacks[endStack].push(temp); // push temp onto the user selected endstack
+   }
 
-}
+    // penguins are the best (wmy ife requested something about penguins)
```
Comment (pieroapretto) [inline apps/04TowersOfHanoi.js:33]: They are the best, yes.

File: apps/04TowersOfHanoi.js [hunk 7]
```diff
@@ -13,30 +13,73 @@ var stacks = {
...
+  var start = stacks[startStack].length; // store length of startStack in start;
+  if(stacks[endStack].length === 0 || stacks[startStack][start - 1] < stacks[endStack][end - 1]){
+    legalMoveCount++; // add 1 to legal move counter here
+    return true;
+  
+}
+  else{
+   // send invalid moves back
+    var temp = stacks[endStack].pop();
+    stacks[startStack].push(temp);
```
Comment (pieroapretto) [inline apps/04TowersOfHanoi.js:51]: If you want to "pop back a bad move", you need to call `movePiece()` before `isLegal()`.

You can avoid all this by only excecuting `movePiece(`) IF `isLegal()` returns true.

File: apps/04TowersOfHanoi.js [hunk 8]
```diff
@@ -13,30 +13,73 @@ var stacks = {
...
+  }
 }
 
-function checkForWin() {
+function checkForWin(){
   // Your code here
-
+/* checks if the length of stacks 'b' or 'c' is 4, or, a completed game, provided all moves
+  were legal */
+if(stacks['b'].length === 4 || stacks['c'].length === 4){
```
Comment (jw56578) [inline apps/04TowersOfHanoi.js:62]: Lets make sure we have proper indention. Make sure everything within curly braces is indented over 2 spaces and that everything is aligned.

File: apps/08jquery-towers-of-hanoi/script.js [hunk 9]
```diff
@@ -1,5 +1,62 @@
...
+// 	if(lastblockofthestack === 0 || datablockattribute < blockdataattribute){
+// 	  $(this).append(block);
+// 	  block = null;
+// 	}
+//       }
+//  	else{
+// 	block = $(this).children().last().deatch();
+// 	}
+// }
+   /*I kept the original code here below*/
```
Comment (jw56578) [inline apps/08jquery-towers-of-hanoi/script.js:31]: Great comments explaining exactly what is happening

**Pull request closed without merging**